lib/ui/semantics.dart

```diff
@@ -386,9 +398,20 @@ class SemanticsFlag {
   /// Must be false when the checkbox is either checked or unchecked.
   static const SemanticsFlag isCheckStateMixed = SemanticsFlag._(_kIsCheckStateMixedIndex, 'isCheckStateMixed');
 
+  /// The semantics node has the quality of either being "selected" or "unselected".
+  ///
+  /// Whether the widget corresponding to this node is currently selected or not
+  /// is determined by the [isSelected] flag.
+  ///
+  /// When this flag is not set, the corresponding widget cannot be selected by
```

Comment:
Actually, if hasSelectedState is a noop in mobile, should we add comments this flag value is only used in web?
Or do you mean in mobile engine we also check the hasSelectedState flag value before setting  hasSelectedState?

Reply:
Oh I don't see an issue with framework always provide`hasSelectedState` for selectable widgets. 

I just wanted to confirm the flag is also used in mobile engines because you previously said it will be mostly noop on mobile.

I just to make sure 
 like this sample in the original issue (https://github.com/flutter/flutter/issues/66673)

```
Semantics(
  selected: true,
  button: true,
  child: Center(
    child: Text(
      'Text',
    ),
  ),
)
```

before the changes, it will be displayed as a selected widget on mobile, but non-selectable widget on web.

After the changes (this PR and future related PR), it will be a non-selectable widget on both mobile and web because now you need to always specify the hasSelectedState flag even for mobile.

Reply:
Talked with yegor offline and this is resolved.
cases like 
```
Semantics(
  selected: true,
  button: true,
  child: Center(
    child: Text(
      'Text',
    ),
  ),
)
```
should automatically set `hasSelectedState` to true